### PR TITLE
Add ansible-lint GHA and format role

### PR DESCRIPTION
**Pull request by konstruktoid**
This PR:
- adds the ansible-lint GitHub action, see https://github.com/ansible/ansible-lint-action
- lints and formats the `install_ansible_rulebook` role

### CODE REVIEW

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by ddonahue007 — APPROVED** [review]
lgtm

**Comment by konstruktoid** [comment]
I'll solve the conflicts and rebase later tonight

**Comment by konstruktoid** [comment]
rebased

File: roles/install_ansible_rulebook/tasks/install_macosx.yml [hunk 1]
```diff
@@ -18,42 +21,49 @@
...
   become: true
 
-- shell: /usr/libexec/java_home
+- name: Register java_home
+  ansible.builtin.command: /usr/libexec/java_home
   register: java_home
+  changed_when: java_home.rc != 0
+  failed_when: java_home.rc != 0
 
 - name: set user envvars
```
Comment (konstruktoid) [inline roles/install_ansible_rulebook/tasks/install_macosx.yml:30]: Can you trigger a test run? That task name has been updated, and I don't know why it checks old code.
```sh
$ git grep envvars roles/
roles/install_ansible_rulebook/tasks/install_fedora.yml:- name: Set user envvars
roles/install_ansible_rulebook/tasks/install_macosx.yml:- name: Set user envvars
```

Comment (ttuffin) [reply roles/install_ansible_rulebook/tasks/install_macosx.yml:30]: Just need to add a couple of newlines and it should be good to go.
https://github.com/ansible/event-driven-ansible/actions/runs/3381227200/jobs/5621457845

Comment (konstruktoid) [reply roles/install_ansible_rulebook/tasks/install_macosx.yml:30]: yeah, but the annoying part is that it isn't triggered locally...
```sh
$ ansible-lint roles/install_ansible_rulebook/

Passed with production profile: 0 failure(s), 0 warning(s) on 11 files.
```

**Comment by konstruktoid** [comment]
all checks have passed, but so sorry for such a messy PR

**Review by ttuffin — APPROVED** [review]
All good :) Thanks for the PR!

**Review by Alex-Izquierdo — APPROVED** [review]

**Pull request merged**